llvm/include/llvm/CodeGen/AccelTable.h

```diff
@@ -310,17 +344,19 @@ void emitAppleAccelTable(AsmPrinter *Asm, AccelTable<DataT> &Contents,
 
 void emitDWARF5AccelTable(AsmPrinter *Asm, DWARF5AccelTable &Contents,
                           const DwarfDebug &DD,
-                          ArrayRef<std::unique_ptr<DwarfCompileUnit>> CUs);
-
+                          ArrayRef<std::unique_ptr<DwarfCompileUnit>> CUs,
+                          TUVectorTy TUSymbols);
+using GetIndexForEntryReturnType =
```

Comment:
this could probably use a better name that describes what's in it - maybe it could be a struct with two named members, and the std::optional could wrap that struct?

Reply:
Name improved?

Reply:
The reason I went with using is because if down the road underlying type changes we don't need to codemod it all over the place. So easier to refactor if the need arises. Maybe you can suggest a better name? :)

Reply:
I think it's few enough places, and more readable as `std::optional<DWARF5AccelTableEntry>` that that's probably the right choice. Overuse of typedefs can make code harder to read/understand/work with. If we remove the optionality/change it to some other mechanism in the future, I think it'll benefit readers to know it isn't optional and whatever else it is when we change it.